Content.Server/Replays/ReplayEvent.cs

```diff
@@ -0,0 +1,220 @@
+﻿using System.Numerics;
+using Content.Shared.Mobs;
+
+namespace Content.Server.Replays;
+
+/// <summary>
+/// Represents an event that occured during gameplay, such as a chat message, a player joining, etc.
+/// with the goal of being able to reconstruct a round without actually running the replay (so external tools parsing it, without knowing the specifics of the fork)
+/// </summary>
+[Serializable, DataDefinition]
+public partial class ReplayEvent
+{
+    [DataField]
+    public double? Time;
+
+    [DataField]
+    public ReplayEventSeverity? Severity;
+
+    [DataField]
+    public ReplayEventType? EventType;
+
+    [DataField]
+    public string? NearestBeacon;
+
+    [DataField]
+    public Vector2? Position;
+
+    [DataField]
+    public string? Map;
```

Comment:
These should be combined into a single field (with a new type) that can nicely represent a location.